examples/ip_tracker/Application.cpp

```diff
@@ -0,0 +1,336 @@
+/**
+ * Copyright 2022 WolkAbout Technology s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "core/utilities/Logger.h"
+#include "core/utilities/Timer.h"
+#include "wolk/WolkBuilder.h"
+#include "wolk/WolkSingle.h"
+
+#include <arpa/inet.h>
+#include <fstream>
+#include <ifaddrs.h>
+#include <iostream>
+#include <netdb.h>
+#include <random>
+#include <string>
+#include <vector>
+
+// Synchronization tools
+std::mutex mutex;
+std::condition_variable conditionVariable;
+
+// returns string from LogLevel
+std::string toString(wolkabout::LogLevel level)
+{
+    if (level == wolkabout::LogLevel::INFO)
+    {
+        return "INFO";
+    }
+    else if (level == wolkabout::LogLevel::WARN)
+    {
+        return "WARN";
+    }
+    else if (level == wolkabout::LogLevel::ERROR)
+    {
+        return "ERROR";
+    }
+    else if (level == wolkabout::LogLevel::DEBUG)
+    {
+        return "DEBUG";
+    }
+    else if (level == wolkabout::LogLevel::TRACE)
+    {
+        return "TRACE";
+    }
+    else if (level == wolkabout::LogLevel::OFF)
+    {
+        return "OFF";
+    }
+    else
+    {
+        return "";
+    }
+}
+// returns LogLevel and if string is valid LogLevel
+std::pair<wolkabout::LogLevel, bool> fromString(std::string in)
+{
+    if (in == "INFO")
+    {
+        return std::make_pair(wolkabout::LogLevel::INFO, true);
+    }
+    else if (in == "WARN")
+    {
+        return std::make_pair(wolkabout::LogLevel::WARN, true);
+    }
+    else if (in == "ERROR")
+    {
+        return std::make_pair(wolkabout::LogLevel::ERROR, true);
+    }
+    else if (in == "DEBUG")
+    {
+        return std::make_pair(wolkabout::LogLevel::DEBUG, true);
+    }
+    else if (in == "TRACE")
+    {
+        return std::make_pair(wolkabout::LogLevel::TRACE, true);
+    }
+    else if (in == "OFF")
+    {
+        return std::make_pair(wolkabout::LogLevel::OFF, true);
+    }
+    else
+    {
+        return std::make_pair(wolkabout::LogLevel::INFO, false);
+    }
+}
+// Class for handling recieved data from feed
+class DeviceDataChangeHandler : public wolkabout::connect::FeedUpdateHandler
+{
+public:
+    DeviceDataChangeHandler(std::string log, bool valid) : m_previousLog(log), m_isLogValid(valid) {}
```

Comment:
m_previousLog represents the actual state of the log, so m_logLevel would be a more fitting name